MisticaCatalog/Source/Catalog/Mistica/Components/UICatalogTextLinkViewController.swift

```diff
@@ -0,0 +1,178 @@
+        super.viewDidLoad()
+
+        view.addSubview(withDefaultConstraints: tableView)
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.keyboardDismissMode = .interactive
+
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard(_:)))
+        tapGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(tapGesture)
+    }
+
+    @objc func dismissKeyboard(_: UITapGestureRecognizer) {
+        cells.flatMap { $0 }.forEach { cell in
+            guard let textFieldCell = cell as? UITextFieldTableViewCell else { return }
+            textFieldCell.textField.resignFirstResponder()
+        }
+    }
+
+    @objc private func textDidChange() {
+        tableView.reloadSections(IndexSet(integer: Section.example.rawValue), with: .automatic)
+    }
+}
+
+extension UICatalogTextLinkViewController: UITableViewDataSource, UITableViewDelegate {
+    func numberOfSections(in _: UITableView) -> Int {
+        Section.allCases.count
+    }
+
+    func tableView(_: UITableView, titleForHeaderInSection section: Int) -> String? {
+        Section(rawValue: section)!.headerTitle
+    }
+
+    func tableView(_: UITableView, numberOfRowsInSection section: Int) -> Int {
+        1
+    }
+
+    func tableView(_: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let section = Section(rawValue: indexPath.section)!
+        switch section {
+        case .example:
+            let textLink = TextLink(fullText: text ?? "", linkedWords: links)
+            textLink.linkDelegate = self
+            return CellView(textLink: textLink, reuseIdentifier: "TextLink")
+        default:
+            return cells[indexPath.section][indexPath.row]
+        }
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard indexPath.section == cells.indices.last!, indexPath.row == cells[indexPath.section].indices.last! else { return }
+        tableView.deselectRow(animated: true)
+        view.endEditing(true)
+    }
+}
+
+extension UICatalogTextLinkViewController: TextLinkDelegate {
+    func textLink(_ view: TextLink, tappedText: String) {
+        let config = SnackbarConfig(
+            title: tappedText,
+            dismissInterval: .fiveSeconds,
+            forceDismiss: false
+        )
+        SnackbarController.shared.showSnackbar(
+            config: config,
+            style: .info,
+            dismissHandler: { reason in
+                print("\(reason.rawValue)")
+            }
+        )
+    }
+}
+
+private extension UICatalogTextLinkViewController {
+    var text: String? {
+        guard let text = textCell.textField.text, !text.isEmpty else { return nil }
+        return text
+    }
+
+    var links: [String]? {
+        guard let links = linkCell.textField.text, !links.isEmpty else { return nil }
+        let separatedLinks = links.components(separatedBy: ",")
```

Comment:
Minor (only applies to catalog: Trim to avoid underline spaces for example in "Tap, here"
```suggestion
        let separatedLinks = links.components(separatedBy: ",").map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
```